src/job_token/windows.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    ffi::{CString, OsString},
+    io, ptr,
+};
+
+use crate::windows_sys::{
+    OpenSemaphoreA, ReleaseSemaphore, WaitForSingleObject, FALSE, HANDLE, SEMAPHORE_MODIFY_STATE,
+    THREAD_SYNCHRONIZE, WAIT_ABANDONED, WAIT_FAILED, WAIT_OBJECT_0, WAIT_TIMEOUT,
+};
+
+const WAIT_ABANDOEND_ERR_MSG: &str = r#" The specified object is a mutex object that was not released by the thread that owned the mutex object before the owning thread terminated. Ownership of the mutex object is granted to the calling thread and the mutex state is set to nonsignaled.
+
+If the mutex was protecting persistent state information, you should check it for consistency."#;
+
+pub(super) struct JobServerClient {
+    sem: HANDLE,
+}
+
+unsafe impl Sync for JobServerClient {}
+unsafe impl Send for JobServerClient {}
+
+impl JobServerClient {
+    pub(super) unsafe fn open(var: OsString) -> Option<Self> {
+        let s = var
+            .to_str()?
+            .split_ascii_whitespace()
+            .filter_map(|arg| {
+                arg.strip_prefix("--jobserver-fds=")
+                    .or_else(|| arg.strip_prefix("--jobserver-auth="))
+            })
+            .find(|s| !s.is_empty())?;
+
+        let name = CString::new(s).ok()?;
+
+        let sem = OpenSemaphoreA(
```

Comment:
I wonder if it's worth complaining if the semaphore name isn't ascii (since we aren't using OpenSemaphoreW). It's hard to imagine that'd ever be the case in reality though.

Reply:
hmm I can switch to that version for maximum compatibikity

Reply:
Since `jobserver` and `jobslot` also uses `OpenSemaphoreA`, I agree that it's a good idea to just check for ascii.

Switching to `OpenSemaphoreW` would make the code a bit harder to maintain since now it differs from upstream in more ways and since `OpenSemaphoreA` already works, it probably doesn't matter.

Reply:
Done.

Reply:
Yes, since the name is almost always auto-generated by `make`, or `cargo` using `jobserver`.